app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,380 @@
+import org.kiwix.kiwixmobile.core.extensions.toast
+import org.kiwix.kiwixmobile.core.utils.dialog.AlertDialogShower
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import org.kiwix.kiwixmobile.localFileTransfer.WifiDirectManager.Companion.getDeviceStatus
+import org.kiwix.kiwixmobile.localFileTransfer.adapter.WifiP2pDelegate
+import org.kiwix.kiwixmobile.localFileTransfer.adapter.WifiPeerListAdapter
+import java.util.ArrayList
+import javax.inject.Inject
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using [PeerGroupHandshakeAsyncTask]
+ * 2) After handshake, starting the files transfer using [SenderDeviceAsyncTask] on the sender
+ * device and [ReceiverDeviceAsyncTask] files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning", "Registered")
+class LocalFileTransferActivity : BaseActivity(),
+  WifiDirectManager.Callbacks {
+  @Inject
+  lateinit var alertDialogShower: AlertDialogShower
+
+  @Inject
+  lateinit var wifiDirectManager: WifiDirectManager
+
+  @Inject
+  lateinit var locationManager: LocationManager
+
+  private var fileListAdapter: FileListAdapter? = null
+  private var wifiPeerListAdapter: WifiPeerListAdapter? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_local_file_transfer)
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    val filesIntent = intent
+    val fileUriArrayList: ArrayList<Uri>? =
+      filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM)
+    val fileForTransfer = fileUriArrayList?.map(::FileItem) ?: emptyList()
+
+    val toolbar: Toolbar =
+      findViewById(R.id.toolbar)
+    setSupportActionBar(toolbar)
+    toolbar.setNavigationIcon(R.drawable.ic_close_white_24dp)
+    toolbar.setNavigationOnClickListener { finish() }
+    wifiPeerListAdapter = WifiPeerListAdapter(
+      WifiP2pDelegate(wifiDirectManager::sendToDevice)
+    )
+
+    list_peer_devices.adapter = wifiPeerListAdapter
+    list_peer_devices.layoutManager = LinearLayoutManager(this)
+    list_peer_devices.setHasFixedSize(true)
+
+    if (fileUriArrayList?.isNotEmpty() == true) {
```

Comment:
I think this `if` isn't even necessary, whats the risk? we display an empty list as opposed to showing nothing? Pretty sure the 2 things are identical.

In the scenario where the if was necessary I also would have used `fileForTransfer` instead, you already processed the unsafe data into this new null safe form.